### PR TITLE
lint: add tests for importShadow

### PR DESCRIPTION
**Pull request by Quasilyte**
Updates #304

### CODE REVIEW

**Comment by Quasilyte** [comment]
@NexoMichael, would you like to do a review for this one?

File: cmd/gocritic/testdata/importShadow/negative_tests.go [hunk 1]
```diff
@@ -10,3 +10,10 @@ import (
 func noWarnings() {
 	fmt.Printf("Hello PI=%v, Rule=%v", math.Pi, lint.Rule{})
 }
+
+func noShadowByParams(x string, y int) (a string, b int) { return }
+
+type noShadow struct{}
+
+/// shadow of imported package 'fmt'
```
Comment (cristaloleg) [inline cmd/gocritic/testdata/importShadow/negative_tests.go]: Do we need this xomment here?

Comment (Quasilyte) [reply cmd/gocritic/testdata/importShadow/negative_tests.go]: No, good catch. Thanks!

**Comment by Quasilyte** [comment]
PTAL.

**Pull request merged**